vehicle/OVMS.V3/components/vehicle_nissanleaf/src/vehicle_nissanleaf.cpp

```diff
@@ -1175,6 +1173,12 @@ void OvmsVehicleNissanLeaf::IncomingFrameCan1(CAN_frame_t* p_frame)
         m_climate_remoteheat->SetValue(d[1] == 0x4b);
         m_climate_remotecool->SetValue(d[1] == 0x71);
         hvac_calculated = (climate_on & (m_climate_fan_speed->AsInt() != 0));
+        //dirty fix until solution is found:
```

Comment:
Would using CC_ACStatus in 0x54C help? I just found it and added it to the database file: https://github.com/dalathegreat/leaf_can_bus_messages/commit/c36252dd65e5fc8eca57fc8f8bc7b40b3354ad5c#diff-433d4c98e1bbbbe20be37efde34e858531adebe851382b0d331c13c000998450

Reply:
Hm, I surely remember dissecting the 54c byte by byte down to bits lol, but you might be right. Will try it out.
![image](https://user-images.githubusercontent.com/38353835/148414033-5d868fa8-36d5-4acf-bc27-c377622c94b4.png)

Reply:
Ok @dalathegreat, so now I remembered that this logic is not working while charging:
```
54c byte 2:
10010000 0x90 <-- preheat ON
11111000 0xf8 <-- preheat OFF + 
00000000 0x00 <-- preheat OFF + charging
```
 that's why I didn't use it form the start. However, I managed to make a simpler logic for this to work. So will push an update soon together with some other tweaks and cleanups.